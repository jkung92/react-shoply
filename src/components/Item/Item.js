import React, { Component } from 'react';
// import { increment } from '../../actionCreators';
//import './Item.css';
// import styled from 'styled-components';

class Item extends Component {
  handleIncrement = () => {
    console.log(`Props in item`, this.props);
    this.props.increment(this.props);
  };

  handleDecrement = () => {
    this.props.decrement(this.props);
  };

  render() {
    return (
      <div
        className="card m-3"
        style={{ width: '18rem', display: 'inline-block' }}
      >
        <img
          src={this.props.image}
          className="card-img-top"
          alt="..."
          style={{ width: '250px', height: '250px' }}
        />
        <div className="card-body">
          <h5 className="card-title">{this.props.name}</h5>
          <p className="card-text">${this.props.price}</p>
          <span style={{ fontSize: '2em', color: 'green' }}>
            <i onClick={this.handleIncrement} className="fas fa-cart-plus" />
          </span>
          <span style={{ fontSize: '2em', color: 'red' }}>
            <i
              onClick={this.handleDecrement}
              className="fas fa-cart-arrow-down"
            />
          </span>
        </div>
      </div>
    );
  }
}

Item.propTypes = {};

Item.defaultProps = {};

export default Item;
