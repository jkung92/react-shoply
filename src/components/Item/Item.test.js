import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import Item from './Item';

it('renders without crashing', function() {
  shallow(<Item />);
});

it('matches snapshot', function() {
  const wrapper = shallow(<Item />);
  const serialized = toJson(wrapper);
  expect(serialized).toMatchSnapshot();
});
