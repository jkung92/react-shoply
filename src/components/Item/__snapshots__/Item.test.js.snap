// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div
  className="Item"
>
  <div
    style={
      Object {
        "display": "flex",
        "justifyContent": "between",
      }
    }
  >
    <h3>
       
       
    </h3>
    <p>
       $
    </p>
    <img
      alt="item.png"
      style={
        Object {
          "height": "250px",
          "width": "250px",
        }
      }
    />
    <span
      style={
        Object {
          "color": "green",
          "fontSize": "2em",
        }
      }
    >
      <i
        className="fas fa-cart-plus"
        onClick={[Function]}
      />
    </span>
    <span
      style={
        Object {
          "color": "red",
          "fontSize": "2em",
        }
      }
    >
      <i
        className="fas fa-cart-arrow-down"
        onClick={[Function]}
      />
    </span>
  </div>
</div>
`;
